File: C++/First Steps in Coding - Basics of C++/Nested Loops/Sum of Two Numbers.cpp
#include <iostream>
using namespace std;
int main() {
	int beg, end, magicNumber, combNum=0;
	bool foundComb = false;
	cin >> beg; cin >> end; cin >> magicNumber;
	for (int k = beg; k <= end; k++) {
		for (int j = beg; j <= end; j++) {
			if (k + j == magicNumber) {
				combNum += 1;
				cout << "Combination N:" << combNum << " (" << k << " + " << j << " = " << magicNumber << ")";
				foundComb = true;
				break;
			}
			else {
				combNum += 1;
			}
		}
		if (foundComb == true) {
			break;
		}
	}
	if (foundComb == false) {
		cout << combNum << " combinations - neither equals " << magicNumber;
	}
}

File: Python/Basics/Nested Loops/Lab/Sum Of Two Numbers.py
a = int(input())
b = int(input())
magicN = int(input())
n1 = a
n2 = a
combinationN = 0
found = False

while n1 <= b:
    while n2 <= b:
        combinationN += 1
        if n1 + n2 == magicN:
            print(f"Combination N:{combinationN} ({n1} + {n2} = {magicN})")
            found = True
            break
        n2 += 1

    if found:
        break
    else:
        n2 = a
        n1 += 1

if not found:
    print(f"{combinationN} combinations - neither equals {magicN}")